### PR TITLE
Remove is_proto check in send_feedback()

### PR DESCRIPTION
**Pull request by matthewphsmith**
What this PR does / why we need it:
By type definition of the request parameter in the input signature, is_proto was always incorrectly set to False. This resulted in handle_raw_custom_metrics acting as a no-op on implementations of send_feedback which matched specification.

This PR removes the isinstance check on the request parameter since it is always true unless the caller is not respecting argument types. If the caller is not respecting argument types, the method would crash anyway upon calling seldon_metrics.update_reward(request.reward) thus I suspect there is no need to protect against such a scenario for reverse compatibility.

Which issue(s) this PR fixes:
Fixes SeldonIO#2606

Special notes for your reviewer:
I opted not to add complexity to the handle_raw_custom_metrics but it may be worth considering refactoring that method to not use the is_proto flag and merely handle isinstance() checking within the method itself.

Does this PR introduce a user-facing change?:
```
Fixes no-op in emitting custom metrics on send_feedback API calls.
```

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @matthewphsmith. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **sachinvarghese**
You can assign the PR to them by writing `/assign @sachinvarghese` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: python/seldon_core/seldon_methods.py
Comment (axsaucedo) [inline python/seldon_core/seldon_methods.py:180]: Ok this is actualyl a bug. It should be `prediction_pb2.Feedback` instead... could you  try that instead? Happy to add the fix, and rerun the tests, not sure how this wasnt' caught before, we should probably extend coverage of tests for that

**Review by axsaucedo — CHANGES_REQUESTED** [review]
Thank you @matthewphsmith - i just added a comment, I am wondering whether that could fix it as I can now see we had a bug there

**Comment by axsaucedo** [comment]
@matthewphsmith will you be able to have a look at this?

**Comment by axsaucedo** [comment]
Closing in favour of #3394

**Pull request closed without merging**
